gradle-plugin/src/main/kotlin/dev/shreyaspatil/composeCompilerMetricsGenerator/plugin/ReportGenPlugin.kt

```diff
@@ -41,16 +46,99 @@ class ReportGenPlugin : Plugin<Project> {
         val android =
             runCatching {
                 target.extensions.getByType(AndroidComponentsExtension::class.java)
-            }.getOrElse { error("This plugin is only applicable for Android modules") }
+            }.getOrNull()
+        val jvm =
+            runCatching {
+                target.extensions.getByType(KotlinJvmProjectExtension::class.java)
+            }.getOrNull()
+        val multiplatform =
+            runCatching {
+                target.extensions.getByType(KotlinMultiplatformExtension::class.java)
+            }.getOrNull()
+        val composeMultiplatform =
+            runCatching {
+                target.extensions.getByType(ComposeExtension::class.java)
+            }.getOrNull()
```

Comment:
I'll get back here with something if there's anything else by which we can write this in better way. I've did something like this in my another plugin: https://github.com/PatilShreyas/bytemask/blob/v1.0.0-alpha01/gradle-plugin/src/main/java/dev/shreyaspatil/bytemask/plugin/BytemaskPlugin.kt#L40-L52

Can we do similar here?

Reply:
Huh, This is basically the way I'm doing it in other plugins too but I tried to follow the same conventions of your plugin, That being said, I will try to rewrite those using `pluginManager.withPlugin`

Reply:
Done 👍